src/features/views/components/ViewColumnDialog/index.tsx

```diff
@@ -1,92 +1,74 @@
 import { useIntl } from 'react-intl';
-import { useMediaQuery } from '@mui/material';
+import { Box, Dialog, useMediaQuery, useTheme } from '@mui/material';
 import { FunctionComponent, useState } from 'react';
 
+import { ColumnChoice } from './choices';
 import ColumnEditor from './ColumnEditor';
 import ColumnGallery from './ColumnGallery';
-import theme from 'theme';
-import ZUIDialog from 'zui/ZUIDialog';
 import {
-  COLUMN_TYPE,
   SelectedViewColumn,
+  ZetkinViewColumn,
 } from 'features/views/components/types';
 
-// These column types will auto save and not open the ColumnEditor
-export const AUTO_SAVE_TYPES = [
-  COLUMN_TYPE.JOURNEY_ASSIGNEE,
-  COLUMN_TYPE.LOCAL_BOOL,
-  COLUMN_TYPE.LOCAL_PERSON,
-  COLUMN_TYPE.PERSON_NOTES,
-];
-
 interface ViewColumnDialogProps {
-  selectedColumn: SelectedViewColumn;
+  columns: ZetkinViewColumn[];
   onCancel: () => void;
-  onSave: (colSpec: SelectedViewColumn) => Promise<void>;
+  onSave: (columns: SelectedViewColumn[]) => Promise<void>;
 }
 
 const ViewColumnDialog: FunctionComponent<ViewColumnDialogProps> = ({
-  selectedColumn,
+  columns: existingColumns,
   onCancel,
   onSave,
 }) => {
-  const isMobile = useMediaQuery(theme.breakpoints.down('md'));
   const intl = useIntl();
-  const [column, setColumn] = useState<SelectedViewColumn>(
-    selectedColumn || {}
-  );
 
-  const onSelectType = (type: COLUMN_TYPE) => {
-    if (AUTO_SAVE_TYPES.includes(type)) {
-      // Save column if no configuration needed
-      onSave({
-        title: intl.formatMessage({
-          id: `misc.views.defaultColumnTitles.${type}`,
-        }),
-        type,
-      });
-      return;
-    }
-    // Create Pending state for column
-    setColumn({
-      config: {},
-      title: '',
-      type,
-    });
+  const theme = useTheme();
+  const fullScreen = useMediaQuery(theme.breakpoints.down('md'));
```

Comment:
Love this! Probably something we should make part of all modals.

Reply:
it's stolen from the ZuiDialog i think? so credit to whoever created that!